interface/src/error.rs

```diff
@@ -1,14 +1,18 @@
 //! Interface error types
 
-use {
-    solana_decode_error::DecodeError,
-    solana_msg::msg,
-    solana_program_error::{PrintProgramError, ProgramError},
-};
+use solana_program_error::{ProgramError, ToStr};
 
 /// Errors that may be returned by the interface.
 #[repr(u32)]
-#[derive(Clone, Debug, Eq, thiserror::Error, num_derive::FromPrimitive, PartialEq)]
+#[derive(
+    Clone,
+    Debug,
+    Eq,
+    thiserror::Error,
+    num_derive::FromPrimitive,
+    num_enum::TryFromPrimitive,
```

Comment:
I'm not sure if we want to generally recommend that people use this instead. Since `ToStr` requires that the enum implements `TryFrom<u32>`, this seemed like the easiest option.

If so, we should add that to the help text in the deprecation warning

Reply:
Agree, I think is a good idea to provide a recommendation. I wonder if we add something to the docs of the `ToStr` with an example will be easier to provide more context.